tokio/src/net/tcp/socket.rs

```diff
@@ -312,14 +320,14 @@ impl TcpSocket {
     ///
     /// [`set_send_buffer_size`]: #method.set_send_buffer_size
     pub fn send_buffer_size(&self) -> io::Result<u32> {
-        self.inner.get_send_buffer_size()
+        self.inner.send_buffer_size().map(|n| n as u32)
```

Comment:
There are several casts in this PR. Do we risk them truncating any data here?

Reply:
There are 5 casts:
- set_send_buffer_size,set_recv_buffer_size (cast u32 as usize): on tokio, [usize is always at least 32 bits](https://github.com/tokio-rs/tokio/blob/cf3206842c0d94ecdaaeb421a58b1c963b627c3d/tokio/src/lib.rs#L357-L368).
- send_buffer_size,recv_buffer_size (cast usize as u32): socket2 gets these values as  c_int ([always i32](https://grep.app/search?q=type%20c_int%20%3D&filter[repo][0]=rust-lang/libc)) and then cast them to usize ([1](https://github.com/rust-lang/socket2/blob/cddb67b8534a46a372beb9fd63c5ce8d1473cbda/src/socket.rs#L857-L858), [2](https://github.com/rust-lang/socket2/blob/cddb67b8534a46a372beb9fd63c5ce8d1473cbda/src/socket.rs#L928-L929)).
- listen (cast u32 as i32): It seems [using cast is wrong](https://github.com/tokio-rs/mio/blob/77bcab0fe57343113700cbb0af76de44ab1ffadf/src/sys/unix/tcp.rs#L37), I'll fix this.

Reply:
fixed in https://github.com/tokio-rs/tokio/pull/4270/commits/5326df1888392e3240d8142e05b17c94c1001350